### DISCUSSION TITLE
Why operator delete pod when change replicas？

### DISCUSSION THREAD

**Discussion by zillaer** [strimzi/strimzi-kafka-operator#6194] [Q&A] [answered]
Why operator delete pod when change replicas？

**Comment by scholzj** [comment, marked as answer]
I think there are several existing discussions about this. Have you tried to search the discussions and issues?

**Reply by zillaer** [reply]
I searched but  still couldn't find it.

When I used _kubectl patch_ to increase the number of STS replicas, the existing replicas were not updated by rolling.

It should be the same when I increase the number of kafka replicas

**Reply by scholzj** [reply]
Well, for example here: https://github.com/strimzi/strimzi-kafka-operator/issues/4913

**Reply by sknot-rh** [reply]
If you update replicas directly in the STS, the number of pods will be reverted next reconciliation.
Strimzi does some extra steps when increasing the number of brokers in the cluster. To keep the cluster working, all the brokers need to be restarted in a specific order (leader is the last).
